pkg/operator/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -227,6 +227,24 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	// predicate functions to filter the events based on the AWS resourceTags presence.
+	infraResourcePredicate := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return hasResourceTags(e.Object)
+		},
+		DeleteFunc: func(event.DeleteEvent) bool {
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return areTagsUpdated(e.ObjectOld, e.ObjectNew)
+		},
+	}
```

Comment:
How often do we expect the Infrastructure resource to be changing? I ask because this is introducing a not-insubstantial amount of AWS-specific code into the generic CredentialsRequest reconciliation file(s). What's the harm in saying "Infrastructure changed...let's requeue everything and make sure any changes to Infrastructure are reflected in the resources this controller is responsible for"?

Reply:
Hi @joelddiaz ,
I couldn't find an answer about the frequency of the changes happening to the Infrastructure resource. Hence, added a predicate to make sure the Reconcile function gets triggered only during the change observed in the resource tags and not for any other change in the Infrastructure resource.